apps/dashboard/src/components/engine/configuration/ip-allowlist.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  useEngineIpAllowlistConfiguration,
+  useEngineSetIpAllowlistConfiguration,
+  useEngineSystemHealth,
+} from "@3rdweb-sdk/react/hooks/useEngine";
+import { Code, Flex, Textarea } from "@chakra-ui/react";
+import { useTxNotifications } from "hooks/useTxNotifications";
+import { isValid } from "ipaddr.js";
+import { useForm } from "react-hook-form";
+import { Button, Heading, Text } from "tw-components";
+
+interface EngineIpAllowlistConfigProps {
+  instanceUrl: string;
+}
+
+interface IpForm {
+  raw: string;
+}
+
+export const EngineIpAllowlistConfig: React.FC<
+  EngineIpAllowlistConfigProps
+> = ({ instanceUrl }) => {
+  const { data: existingIpAllowlist } =
+    useEngineIpAllowlistConfiguration(instanceUrl);
+  const { mutateAsync: setIpAllowlist } =
+    useEngineSetIpAllowlistConfiguration(instanceUrl);
+
+  const { onSuccess, onError } = useTxNotifications(
+    "IP Allowlist updated successfully.",
+    "Failed to update IP Allowlist",
+  );
+
+  const { data: engineHealthInfo } = useEngineSystemHealth(instanceUrl);
+
+  const form = useForm<IpForm>({
+    defaultValues: { raw: existingIpAllowlist?.join("\n") ?? "" },
+  });
+
+  const onSubmit = async (data: IpForm) => {
+    try {
+      const ips = data.raw
+        .split(/[\n,]/)
+        .filter((url) => !!url)
+        .map((ip) => ip.trim());
+
+      // Check if all IPs are well formed.
+      const incorrectIp = ips.find((ip) => !isValid(ip));
+
+      if (incorrectIp) {
+        throw new Error(`Invalid IP Address: ${incorrectIp}`);
+      }
+
+      await setIpAllowlist({ ips });
+      form.reset({
+        raw: ips.join("\n"),
+      });
+      onSuccess();
+    } catch (error) {
+      onError(error);
+    }
+  };
+
+  if (!engineHealthInfo?.features?.includes("IP_ALLOWLIST")) return <></>;
```

Comment:
in this case please simply return `null`

Reply:
also prefer new line for the return

Reply:
made these changes!